crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -38,39 +39,24 @@ impl CumulocityMapper {
 
         Ok(topic_filter)
     }
+}
 
-    pub async fn init_session(&mut self) -> Result<(), anyhow::Error> {
-        info!("Initialize tedge mapper session");
-        mqtt_channel::init_session(&self.get_mqtt_config()?).await?;
-        Ok(())
-    }
-
-    pub async fn clear_session(&mut self) -> Result<(), anyhow::Error> {
-        info!("Clear tedge mapper session");
-        mqtt_channel::clear_session(&self.get_mqtt_config()?).await?;
-        Ok(())
+#[async_trait]
+impl TEdgeComponent for CumulocityMapper {
+    fn session_name(&self) -> &str {
+        CUMULOCITY_MAPPER_NAME
     }
 
-    fn get_mqtt_config(&mut self) -> Result<mqtt_channel::Config, anyhow::Error> {
+    async fn init(&self) -> Result<(), anyhow::Error> {
+        info!("Initialize tedge mapper c8y");
+        create_directories()?;
         let operations = Operations::try_new("/etc/tedge/operations", "c8y")?;
-        let mqtt_topic = Self::subscriptions(&operations)?;
-        let config_repository =
-            tedge_config::TEdgeConfigRepository::new(tedge_config::TEdgeConfigLocation::default());
-        let tedge_config = config_repository.load()?;
-
-        let mqtt_config = mqtt_channel::Config::default()
-            .with_host(tedge_config.query(MqttBindAddressSetting)?.to_string())
-            .with_port(tedge_config.query(MqttPortSetting)?.into())
-            .with_session_name(CUMULOCITY_MAPPER_NAME)
-            .with_clean_session(false)
-            .with_subscriptions(mqtt_topic);
-
-        Ok(mqtt_config)
+        let mqtt_topics = CumulocityMapper::subscriptions(&operations)?;
+        let mqtt_config = self.get_mqtt_config()?;
+        mqtt_channel::init_session(&mqtt_config.with_subscriptions(mqtt_topics)).await?;
```

Comment:
It would be good to introduce a  `TEdgeComponent::init_session` reference implementation and to use it here.

```suggestion
        self.init_session(CumulocityMapper::subscriptions(&operations)?).await?
```